simulator/gdb/sim_wrapper.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * sim_wrapper.cpp - Functional simulator wrappers for GDB
+ * @author Vyacheslav Kompan
+ * Copyright 2018 MIPT-MIPS
+ */
+
+#include "sim_wrapper.h"
+
+/* MIPT-MIPS simulator interfaces */
+#include <simulator.h>
+#include <infra/config/config.h>
+#include <mips/mips.h>
+#include <func_sim/func_sim.h>
+
+#define ISA_NAME "mips32"
+
+typedef struct {
+    std::unique_ptr<Simulator> ptr = nullptr;
+    std::string filename;
+} SimulatorInstance;
+static SimulatorInstance simInstance;
```

Comment:
Your implementation assumes there is only one simulation instance.
If I understand correctly, SIM_DESC is a descriptor of simulator used to uniqually identify it. so several cores may me modeled as well.

Reply:
SIM_DESC is a C struct, and I can't keep a pointer to Simulator class in it. However, I think can keep a void pointer to a simulator instance in it, and cast it in C++ code.

Reply:
You may keep an integer descriptor here, and use it as an index to static collection of instances.